source/common/network/socket_impl.h

```diff
@@ -93,6 +100,7 @@ class ConnectionInfoSetterImpl : public ConnectionInfoSetter {
   std::string ja3_hash_;
   absl::optional<std::chrono::milliseconds> round_trip_time_;
   FilterChainInfoConstSharedPtr filter_chain_info_;
+  ListenerInfoConstSharedPtr listener_info_;
```

Comment:
@kyessenov See, the connection info own the listener info and connection info will owned by the `StreamInfo`. So, I didn't get you concern. Do I missed something?

Reply:
Can `StreamInfo` outlive `ListenerImpl`? Let me try to analyze the ownership, but it's definitely not obvious without knowing the flow control.

Reply:
OK, yeah, the "shared" Listener parts should be carried over by the completion until all workers stop handling streams [here](https://github.com/envoyproxy/envoy/blob/main/source/common/listener_manager/listener_manager_impl.cc#L852). So we don't have to use a shared pointer.